onyx-bootstrap.rb

```diff
@@ -140,57 +170,60 @@ def main!
 		require_gem 'nokogiri', true
-		puts "  Loading webpage" if $verbose
+		puts "  Done"
+	}]
+	tasks << ["Downloading latest Crystal binary", ->{
+		spout "  Loading webpage"
 		html = openurl "https://github.com/crystal-lang/crystal/releases"
-		puts "  Parsing html" if $verbose
+		spout "  Parsing html"
 		html = Nokogiri::HTML(html)
-		puts "  Scanning html" if $verbose
+		spout "  Scanning html"
 		href = "https://github.com" + (html/"ul[class='release-downloads']/li/a").select { |a| (a/"strong").inner_text =~ /crystal-(.*)-linux-#{$bittedness}.tar.gz/ }.first[:href]
 		raise "Could not find latest Crystal release on github" unless href
-		puts "  Found Crystal release: " + href if $verbose
-		puts "  Downloading archive" if $verbose
+		spout "  Found Crystal release: " + href, "  Downloading archive"
 		Dir.chdir $tempdir do
 			execute %{wget#{' -q' unless $verbose} "#{href}" -O - | tar zx}
 		end
 	}]
 	tasks << ["Checking if Crystal works", ->{
 		begin
 			crystal = (Pathname(Dir.glob($tempdir + 'crystal-*')[0]) + 'bin/crystal').to_s
-			execute "echo 'puts' | #{crystal} >/dev/null 2>/dev/null eval", false
+			execute "echo 'puts' | #{crystal} >/dev/null 2>/dev/null eval"
 			puts "  Yes it does. Will wonders never cease?"
 		rescue
+			binding.pry
 			puts "  Of course not. That would be too easy."
 			tasks.unshift ["Installing Crystal prerequisites", ->{
 				workingdir = Pathname(Dir.mktmpdir "crystal-env")
 				begin
 					Dir.chdir workingdir do
-						puts "  Created temporary directory: #{workingdir}" if $verbose
+						spout "  Created temporary directory: #{workingdir}"
 						# $logger << "Current directory is ⟨W%s⟩. Contents: %s" % [Dir.current, Dir.entries(Dir.current).wrap("⟨W", "⟩").join(", ")]
 						puts $verbose ? "  Downloading bdwgc" : "  Downloading and compiling bdwgc"
 						execute "git clone git://github.com/ivmai/bdwgc.git"
-						puts "  Compiling bdwgc" if $verbose
+						spout "  Compiling bdwgc"
 						Dir.chdir "bdwgc" do
 							execute "git clone git://github.com/ivmai/libatomic_ops.git"
 							execute "autoreconf -vif"
 							execute "automake --add-missing"
 							execute "./configure"
 							execute "make"
-							execute "make install"
+							execute "make install", sudo: true
 						end
 
-						puts "", $verbose ? "  Downloading pcl" : "  Downloading and compiling pcl"
+						puts $verbose ? "\n  Downloading pcl" : "  Downloading and compiling pcl"
 						execute "curl -O#{' -s' unless $verbose} http://www.xmailserver.org/pcl-1.12.tar.gz"
 						execute "tar -xzf pcl-1.12.tar.gz"
-						puts "  Compiling pcl" if $verbose
+						spout "  Compiling pcl"
 						Dir.chdir "pcl-1.12" do
 							execute "./configure"
 							execute "make"
-							execute "make install"
+							execute "make install", sudo: true
 						end
-						puts ""
-						puts "  Deleting old versions" if $verbose
+						puts "" if $verbose
+						spout "  Deleting old versions"
 						rm_f "/usr/lib/libgc.so*"
```

Comment:
#!$% bollocks. Build and release has always been someone else's problem in my entire coding life, so I'm dumbfounded when it comes to those details. Guess I'll wait with this one, will take me to much time to figure out how to go about such, surely easy if know-how'd, stuff.
A simple Y/N will do for the rm - the less we mess with a users system, the better.